[BUG] Dataset turns to bound with a failed helm installation

**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**
Dataset turns to bound with a failed helm installation. The BUG can be reproduced by the following order:
1. create a dataset & runtime with misconfigs (e.g. A dataset with a very long name)
2. runtime controller attempts to install dataset related resources into K8s cluster, but failed. 

An example is shown below:
```
$ ddc-helm install  -f /tmp/demo-dataset-abcdefghijklmnopqrstuvwxyz-0123456789-jindofsx-values.yaml1228478600 --output json --namespace default demo-dataset-abcdefghijklmnopqrstuvwxyz-0123456789 /charts/jindofsx           
coalesce.go:199: warning: destination for master is a table. Ignoring non-table value <nil>                                                                                                                                           
coalesce.go:199: warning: destination for workersAndClients is a table. Ignoring non-table value <nil>                                                                                                                                
Error: failed pre-install: warning: Hook pre-install jindofs/templates/config/jindofs-client-conf.yaml failed: ConfigMap "demo-dataset-abcdefghijklmnopqrstuvwxyz-0123456789-jindofs-client-config" is invalid: metadata.labels: Inval
id value: "demo-dataset-abcdefghijklmnopqrstuvwxyz-0123456789-jindofs-client-config": must be no more than 63 characters    
```

3. runtime controller retries the reconcilation loop, but found the release still exists.(the helm release is in `pending-install` status) 

```
ddc-helm status -n default demo-dataset-abcdefghijklmnopqrstuvwxyz-0123456789
NAME: demo-dataset-abcdefghijklmnopqrstuvwxyz-0123456789
LAST DEPLOYED: Thu Mar 16 17:08:06 2023
NAMESPACE: default
STATUS: pending-install
REVISION: 1
TEST SUITE: None
```

4. runtime controller mistakenly considers the release is already installed, so just go on and update Dataset's phase to `Bound`



**What you expect to happen:**
Dataset should not turn to `bound` phase if the Fluid fails to install kubernetes resources.

**How to reproduce it**
Following the steps mentioned above. Just use a dataset with a very long name can reproduce it.


**Additional Information**

--- a/pkg/utils/helm/utils.go
+++ b/pkg/utils/helm/utils.go
@@ -25,6 +25,7 @@ import (
 	"syscall"
 
 	"github.com/fluid-cloudnative/fluid/pkg/utils"
+	"github.com/pkg/errors"
 )
 
 // InstallRelease installs the release with cmd: helm install -f values.yaml chart_name, support helm v3
@@ -61,7 +62,13 @@ func InstallRelease(name string, namespace string, valueFile string, chartName s
 
 	if err != nil {
 		log.Error(err, "failed to execute InstallRelease() command", "command", cmd.String())
-		return fmt.Errorf("failed to create engine-related kubernetes resources")
+		err = fmt.Errorf("failed to install kubernetes resources of %s", chartName)
+
+		rollbackErr := DeleteReleaseIfExists(name, namespace)
+		if rollbackErr != nil {
+			log.Error(err, "failed to rollback installed helm release after InstallRelease() failure", "name", name, "namespace", namespace)
+		}
+		return err
 	}
 
 	return nil
@@ -87,7 +94,7 @@ func CheckRelease(name, namespace string) (exist bool, err error) {
 		return exist, err
 	}
 
-	err = cmd.Wait()
+	resultBytes, err := cmd.CombinedOutput()
 	if err != nil {
 		if exiterr, ok := err.(*exec.ExitError); ok {
 			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
@@ -104,7 +111,27 @@ func CheckRelease(name, namespace string) (exist bool, err error) {
 	} else {
 		waitStatus := cmd.ProcessState.Sys().(syscall.WaitStatus)
 		if waitStatus.ExitStatus() == 0 {
-			exist = true
+			// ####### helm status output ######
+			// NAME: demo-dataset
+			// LAST DEPLOYED: Thu Mar 16 17:08:06 2023
+			// NAMESPACE: default
+			// STATUS: deployed
+			// REVISION: 1
+			// TEST SUITE: None
+			// ####### END #######
+			resultLines := strings.Split(string(resultBytes), "\n")
+			for _, line := range resultLines {
+				if strings.HasPrefix(line, "STATUS: ") {
+					if strings.Replace(line, "STATUS: ", "", 1) == "deployed" {
+						exist = true
+					} else {
+						rollbackErr := DeleteRelease(name, namespace)
+						if rollbackErr != nil {
+							err = errors.Wrapf(rollbackErr, "failed to rollback failed release (namespace: %s, name: %s)", namespace, name)
+						}
+					}
+				}
+			}
 		} else {
 			if waitStatus.ExitStatus() != -1 {
 				return exist, fmt.Errorf("unexpected return code %d when exec helm status %s -n %s",

--- a/pkg/utils/helm/utils.go
+++ b/pkg/utils/helm/utils.go
@@ -87,12 +87,12 @@ func CheckRelease(name, namespace string) (exist bool, err error) {
 	// 	cmd.Env = append(cmd.Env, fmt.Sprintf("KUBECONFIG=%s", types.KubeConfig))
 	// }
 
-	if err := cmd.Start(); err != nil {
-		// log.Fatalf("cmd.Start: %v", err)
-		// log.Error(err)
-		log.Error(err, "failed to start CheckRelease() command", "command", cmd.String())
-		return exist, err
-	}
+	// if err := cmd.Start(); err != nil {
+	// 	// log.Fatalf("cmd.Start: %v", err)
+	// 	// log.Error(err)
+	// 	log.Error(err, "failed to start CheckRelease() command", "command", cmd.String())
+	// 	return exist, err
+	// }
 
 	resultBytes, err := cmd.CombinedOutput()
 	if err != nil {

--- a/pkg/utils/helm/utils.go
+++ b/pkg/utils/helm/utils.go
@@ -87,13 +87,6 @@ func CheckRelease(name, namespace string) (exist bool, err error) {
 	// 	cmd.Env = append(cmd.Env, fmt.Sprintf("KUBECONFIG=%s", types.KubeConfig))
 	// }
 
-	// if err := cmd.Start(); err != nil {
-	// 	// log.Fatalf("cmd.Start: %v", err)
-	// 	// log.Error(err)
-	// 	log.Error(err, "failed to start CheckRelease() command", "command", cmd.String())
-	// 	return exist, err
-	// }
-
 	resultBytes, err := cmd.CombinedOutput()
 	if err != nil {
 		if exiterr, ok := err.(*exec.ExitError); ok {

